packages/blade/src/components/Tabs/Tabs.native.tsx

```diff
@@ -0,0 +1,214 @@
+
+  const renderTabLabel = React.useCallback(
+    ({ route, focused }) => {
+      const { title, leading, trailing } = route;
+      const selectedState = focused ? 'selected' : 'unselected';
+      // eslint-disable-next-line react-hooks/rules-of-hooks
+      const validatedTrailingComponent = useTabsItemPropRestriction(trailing, size);
+
+      return (
+        <StyledTabButton autoWidth={!autoWidth && !isFilled} variant={variant} size={size}>
+          <Box display="flex" alignItems="center" flexDirection="row" gap="spacing.3">
+            {leading
+              ? React.cloneElement(leading as React.ReactElement, {
+                  size: iconSizeMap[size],
+                  color: `surface.action.icon.default.lowContrast`,
+                })
+              : null}
+            <Text
+              color={textColor[selectedState].default}
+              size={size === 'medium' ? 'medium' : 'large'}
+              weight={size === 'medium' ? 'bold' : 'regular'}
+            >
+              {title}
+            </Text>
+            {validatedTrailingComponent}
+          </Box>
+        </StyledTabButton>
+      );
+    },
+    [autoWidth, isFilled, size, variant],
+  );
+
+  const renderTabBar = React.useCallback(
+    (props) => (
+      <TabBar
+        {...props}
+        gap={0}
+        android_ripple={{ borderless: true, color: 'transparent' }}
+        scrollEnabled={!autoWidth && !isFilled}
+        tabStyle={{
+          padding: 0,
+          margin: 0,
+          minHeight: 0,
+        }}
+        contentContainerStyle={
+          isFilled
+            ? {}
+            : {
+                borderBottomColor: theme.colors.surface.border.normal.lowContrast,
+                borderBottomWidth: 1.5,
+              }
+        }
+        style={{
+          ...(isFilled
+            ? {
+                shadowOpacity: 0,
+                shadowColor: 'transparent',
+                borderRadius: theme.border.radius.small,
+                borderWidth: theme.border.width.thick,
+                borderColor: theme.colors.surface.border.normal.lowContrast,
+                backgroundColor: theme.colors.surface.background.level2.lowContrast,
+                padding: theme.spacing[2],
+              }
+            : {
+                shadowOpacity: 0,
+                shadowColor: 'transparent',
+                backgroundColor: 'transparent',
+              }),
+        }}
+        renderIndicator={TabIndicator}
+        renderLabel={renderTabLabel}
+      />
+    ),
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+    [autoWidth, isFilled, renderTabLabel],
+  );
+
+  return (
+    <TabsContext.Provider value={contextValue}>
+      <TabView
+        navigationState={{
+          index,
+          routes,
+        }}
+        renderScene={SafeSceneMap(panels)}
+        renderTabBar={renderTabBar}
+        onIndexChange={setIndex}
+        initialLayout={initialLayout}
+        lazy={false}
```

Comment:
Why not lazy load it? Give it to consumers as an option. On our mobile apps we're using lazy load in places. If we don't, the initial load is very laggy since it makes all the API calls for all the tabs at once.

Reply:
Added isLazy prop so consumers can decide this.